src/gui/mainwindow.cpp

```diff
@@ -312,6 +312,8 @@ Shell* MainWindow::shell()
 void MainWindow::saveWindowGeometry()
 {
 	QSettings settings("nvim-qt", "window-geometry");
+	bool restore_window_geometry = settings.value("restore_window_geometry", true).toBool();
+	settings.setValue("restore_window_geometry", restore_window_geometry);
 	settings.setValue("window_geometry", saveGeometry());
 	settings.setValue("window_state", saveState());
```

Comment:
I think we should avoid writing `window_geometry` and `window_state`  values if `restore_window_geometry` is set to false.

I've seen some users who track their config files in VCS complain about the constant changing of these values every open/close. I'll try to find the specific issue and reference it in the main discussion.